cmd/resourcegen/main.go

```diff
@@ -84,52 +84,53 @@ type config struct {
 	WithHelp bool
 }
 
-func (c config) ResourceLetter() string {
-	return strings.ToLower(string(c.ResourceName[0]))
+func (c config) FullResourceName() string {
+	prefix := c.CloudProvider
+	if c.CloudProvider == "azure" {
+		prefix = "azurerm"
+	}
+
+	return fmt.Sprintf("%s_%s", prefix, c.Filename)
 }
 
-func (c config) CloudProviderTerraformBlock() string {
-	switch c.CloudProvider {
-	case "aws":
-		return `provider "aws" {
-  region                      = "us-east-1"
-  skip_credentials_validation = true
-  skip_metadata_api_check     = true
-  skip_requesting_account_id  = true
-  skip_get_ec2_platforms      = true
-  skip_region_validation      = true
-  access_key                  = "mock_access_key"
-  secret_key                  = "mock_secret_key"
+func (c config) RFuncName() string {
+	return "new" + c.ResourceName
 }
-`
-	case "google":
-		return `provider "google" {
-  credentials = "{\"type\":\"service_account\"}"
-  region      = "us-central1"
+
+func (c config) RegistryFuncName() string {
+	return "get" + c.ResourceName + "RegistryItem"
 }
-`
-	case "azure":
-		return `provider "azurerm" {
-  skip_provider_registration = true
-  features {}
+
+func (c config) ProviderDirectory() string {
+	return path.Join("internal/providers/terraform/", c.CloudProvider)
```

Comment:
```suggestion
	return "internal/providers/terraform/" + c.CloudProvider
```

no need for path.Join as `c.CloudProvider` won't have a slash in the name

Reply:
This is my inner Rubyist tries to avoid string concatenations as much as possible :D If the path string doesn't have last slash, it would look better, I guess :)